packages/migrate/commonsChunkPlugin/commonsChunkPlugin.ts

```diff
@@ -0,0 +1,147 @@
+import {
+	addOrUpdateConfigObject,
+	createIdentifierOrLiteral,
+	createLiteral,
+	createProperty,
+	findAndRemovePluginByName,
+	findPluginsByName,
+} from "@webpack-cli/utils/ast-utils";
+
+import { IJSCodeshift, INode } from "../types/NodePath";
+
+/**
+ *
+ * Transform for CommonsChunkPlugin. If found, removes the
+ * plugin and sets optimizations.namedModules to true
+ *
+ * @param {Object} j - jscodeshift top-level import
+ * @param {Node} ast - jscodeshift ast to transform
+ * @returns {Node} ast - jscodeshift ast
+ */
+export default function(j: IJSCodeshift, ast: INode): INode {
+
+	let pluginProps: INode[];
+	const cacheGroupsArray: INode[] = [];
+
+	// find old options
+	const CommonsChunkPlugin = findPluginsByName(j, ast, ["webpack.CommonsChunkPlugin"]);
+
+	if (!CommonsChunkPlugin.size()) { return ast; }
+
+	CommonsChunkPlugin
+		.forEach((path: INode): void => {
+			pluginProps = path.value.arguments[0].properties; // any node
+		});
+
+	// create chunk cache group option
+	function createChunkCache(chunkName) {
+		const commonProperties = [
+			createProperty(
+				j,
+				"name",
+				chunkName.value,
+			),
+		];
+		switch (chunkName.value) {
+			case "vendor":
+				return j.property(
```

Comment:
this is not always the case, user supplies the regex

Reply:
Can you please give an example syntax of how the user supplies the regex? I can't find it in the [CommonsChunkPlugin docs](https://webpack.js.org/plugins/commons-chunk-plugin/)

Reply:
> If user supplies the test prop they can regex chunks

I can't find that option in the older CommonsChunkPlugin docs. `test` is supported in SplitChunksPlugin docs.

Reply:
No need to take review comments for actual comments. A regex to match a chunk was possible in v.2 if I recall right. @sokra needs to guide you here, I don't fully know backwards compat spec